stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -1465,20 +1469,21 @@ internal constructor(
 
         val relevantPlugins = plugins.filterIsInstance<QueryChannelListener>().also(::logPlugins)
 
-        return api.queryChannel(channelType, channelId, request)
-            .doOnStart(scope) {
-                relevantPlugins.forEach { plugin ->
-                    logger.logD("[queryChannel] #doOnStart; plugin: ${plugin::class.qualifiedName}")
-                    plugin.onQueryChannelRequest(channelType, channelId, request)
-                }
+        return queryChannelsPostponeHelper.postponeCall {
+            api.queryChannel(channelType, channelId, request)
+        }.doOnStart(scope) {
```

Comment:
`api.queryChannel(channelType, channelId, request)` is postponed until WS connection gets established.